src/System.Private.ServiceModel/tests/Stress/SharedPoolsOfWCFObjects/SharedPoolsOfWCFObjects.cs

```diff
@@ -57,33 +65,81 @@ public static void CreateFactoriesAndChannelsUseAllOnceCloseAll()
 
     // One of the most common scenario: create a pool of channel factories, use each factory to create a pool of channels,
     // and then use all pooled channels for the duration of the stress run
-    public class PooledFactoriesAndChannels
+    public class PooledFactories<ChannelType, TestTemplate>
+    where ChannelType : class
+    where TestTemplate : ITestTemplate<ChannelType>, new()
     {
         private static CallStats s_CreateChannelStats = new CallStats();
         private static CallStats s_CallChannelStats = new CallStats();
         private static CallStats s_CloseChannelStats = new CallStats();
         private static CallStats s_CloseFactoryStats = new CallStats();
+
+        private static ITestTemplate<ChannelType> s_test;
+        private static PoolOfThings<ChannelFactory<ChannelType>> s_pooledChannelFactories;
+
+        static PooledFactories()
+        {
+            s_test = new TestTemplate();
+            s_pooledChannelFactories = StaticDisposablesHelper.AddDisposable(
+                new PoolOfThings<ChannelFactory<ChannelType>>(
+                    maxSize: 10,
+                    createInstance: () =>
+                        s_test.CreateChannelFactory(),
+                    destroyInstance: (chf) =>
+                         s_CloseFactoryStats.CallActionAndRecordStats(() => s_test.CloseFactory(chf))));
+        }
+        public static void CreateUseAndCloseChannels()
+        {
+            foreach (var factory in s_pooledChannelFactories.GetAllPooledInstances())
+            {
+                ChannelType channel = null;
+                s_CreateChannelStats.CallActionAndRecordStats(() =>
+                    channel = s_test.CreateChannel(factory));
+
+                if (channel != null)
+                {
+                    s_CallChannelStats.CallActionAndRecordStats(() =>
+                        s_test.UseChannel()(channel));
+
+                    s_CloseChannelStats.CallActionAndRecordStats(() =>
+                        s_test.CloseChannel(channel));
+                }
+            }
+        }
+    }
+
+    public class PooledFactoriesAndChannels<ChannelType, TestTemplate>
+        where ChannelType : class
```

Comment:
Thanks Ron! With this positive encouragement I am also adding TestTemplate generic parameters to allow for running each test with different settings (suitable for a specific perf or stress scenario).